juju/juju
# [JUJU-3447] Trigger a model firewall reconfigure when machines are created

Trigger a model firewall reconfigure when machines are created

This is because models with no machines don't have model security
groups. They can be created when a new machine is, so there are two
events we may wish to configure the model security group
- On change to the config
- On machine creation/start

Change the machine watcher to watch machine agent start times, i.e.
when the machine is actually created

The only reason this worked prior was because the ec2 provider was not
returning the correct errortype, leading to the firewaller bouncing
until the sec group was created. Resolve this by returning a NotFound
error

## Checklist


## QA steps

Verifying the ssh-allow integration test succeeds should be enough
```
./main.sh -v -c aws -p ec2 firewall
```

But also
```
$ juju bootstrap aws/eu-west-2 aws
$ juju add-model m
```
Then verify debug logs for m have no errors, but display the following warning
```
$ juju debug-log
...
controller-0: 12:48:52 WARNING juju.worker.firewaller Model firewall not found so cannot re-write firewall rule ["22/tcp from 0.0.0.0/0,::/0"]. Ignoring
...
```

After seeing this, add a machine
```
juju add-machine
```
And verify in the aws console that it's security group is correctly configured to open port 22 to 0.0.0.0/0 and ::/0
